client/state/data-layer/wpcom-http/index.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * External dependencies
+ */
+import { get, noop } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import wpcom from 'lib/wp';
+import { WPCOM_HTTP_REQUEST } from 'state/action-types';
+import { extendAction } from 'state/utils';
+
+/**
+ * Returns the appropriate fetcher in wpcom given the request method
+ *
+ * fetcherMap :: String -> (Params -> Query -> [Body] -> Promise)
+ *
+ * @param {string} method name of HTTP method for request
+ * @returns {Function} the fetcher
+ */
+const fetcherMap = method => get( {
```

Comment:
There's a couple we're missing here, though I'm not sure we're using these anywhere with the WordPress.com API (more likely we are/will with Core API):

- `put`
- `del`

https://github.com/Automattic/wpcom.js/blob/master/lib/util/request.js